File: projects/JavaFXApps/SnakeFX/src/main/java/net/rickiekarp/snakefx/core/Direction.java
package net.rickiekarp.snakefx.core;

/**
* This enum represents the directions that the snake can go.
*/
public enum Direction {

	UP(false),

	DOWN(false),

	LEFT(true),

	RIGHT(true);

	private boolean horizontal;

	private Direction(final boolean horizontal) {
		this.horizontal = horizontal;
	}

	public boolean hasSameOrientation(final Direction other) {
		if (other == null) {
			return false;
		}
		return (horizontal == other.horizontal);
	}
}

File: projects/JavaFXApps/SnakeFX/src/main/kotlin/net/rickiekarp/snakefx/core/Direction.kt
package net.rickiekarp.snakefx.core

/**
 * This enum represents the directions that the snake can go.
 */
enum class Direction private constructor(private val horizontal: Boolean) {

    UP(false),

    DOWN(false),

    LEFT(true),

    RIGHT(true);

    fun hasSameOrientation(other: Direction?): Boolean {
        return if (other == null) {
            false
        } else horizontal == other.horizontal
    }
}
